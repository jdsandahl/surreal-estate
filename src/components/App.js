import React, { useState } from "react";
import "../styles/App.css";
import { Route, Switch } from "react-router-dom";
import NavBar from "./NavBar.js";
import Properties from "./Properties";
import AddProperty from "./AddProperty";

function App() {
  const [userId, setUserId] = useState("");

  const handleLogin = (response) => {
    const loginDetails = response.id;
    setUserId(loginDetails);
  };

  const handleLogout = () => {
    window.FB.logout(() => setUserId(""));
  }

  return (
    <div className="App">
      <NavBar onLogin={handleLogin} onLogout={handleLogout} userId={userId} />
      <Switch>
        <Route exact path="/" component={Properties} />
        <Route exact path="/add-property" component={AddProperty} />
      </Switch>
      <div id="fb-root"></div>
    </div>
  );
}

export default App;
