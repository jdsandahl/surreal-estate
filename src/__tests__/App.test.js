import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { Router } from "react-router-dom";
import { createMemoryHistory } from 'history'
import App from "../components/App";

test("full app rendering & navigating correctly", () => {
  const history = createMemoryHistory();
  const { container, getByText, getByTestId } = render(
    <Router history={history}>
      <App />
    </Router>
  );
  const titleElement = getByText(/surreal estate/i);

  expect(titleElement).toBeInTheDocument();
  expect(container.innerHTML).toMatch('Properties Page');

  fireEvent.click(getByText(/add property/i));
  expect(getByTestId('AddProperty')).toBeInTheDocument();  
});
