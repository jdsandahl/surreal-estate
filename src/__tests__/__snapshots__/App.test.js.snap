// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App the page renders correctly against snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="navbar"
    >
      <div
        class="burger"
      >
        <input
          class="burger__toggle"
          type="checkbox"
        />
        <div
          class="burger__lines"
        >
          <div />
        </div>
      </div>
      <img
        alt="logo"
        class="logo"
        src="real-estate-logo.png"
      />
      <h2
        class="navbar-title"
      >
        <a
          class="navbar-title-link"
          href="/"
        >
          Surreal Estate
        </a>
      </h2>
      <ul
        class="navbar-links navbar-links--visible"
      >
        <li
          class="navbar-links-item"
        >
          <a
            class="navbar-link"
            href="/properties"
          >
            View Properties
          </a>
        </li>
        <li
          class="navbar-links-item"
        >
          <a
            class="navbar-link"
            href="/add-property"
          >
            Add Property
          </a>
        </li>
        <li
          class="navbar-links-item"
        >
          <p>
            TypeError: Cannot read property 'parentNode' of undefined
          </p>
        </li>
      </ul>
    </div>
    <div
      class="home-page"
    >
      <aside
        class="home-page__welcome"
      >
        <h2
          class="home-page__heading"
        >
          Welcome
        </h2>
        <p>
          From our app you're able to view, search and post properties within the UK.
        </p>
        <p>
          Save your favourite properties to view later by logging in with Facebook
        </p>
      </aside>
    </div>
    <div
      class="footer"
    >
      <ul
        class="footer__contents"
      >
        <li
          class="footer__item"
        >
          Justin Sandahl 2020
        </li>
      </ul>
    </div>
    <div
      id="fb-root"
    />
  </div>
</DocumentFragment>
`;
